util/metrics/registryCommon.go

```diff
@@ -25,11 +26,17 @@ import (
 // Metric represent any collectable metric
 type Metric interface {
 	WriteMetric(buf *strings.Builder, parentLabels string)
-	AddMetric(values map[string]string)
+	AddMetric(values map[string]float64)
 }
 
 // Registry represents a single set of metrics registry
 type Registry struct {
 	metrics   []Metric
 	metricsMu deadlock.Mutex
 }
+
+var sanitizeTelemetryCharactersRegexp = regexp.MustCompile("(^[^a-zA-Z_]|[^a-zA-Z_0-9])")
+
+func sanitizeTelemetryName(name string) string {
```

Comment:
I'd love to see a test for this function. It should be easy to write and would demonstrate what that slightly messy regex is supposed to achieve. And a comment, like, "ensure telemetry name starts with `[a-zA-Z0-9_]`"

Reply:
thanks, I have added a test and comment, and also noticed another test I was modifying was build-tagged out of being run by CI, updated it too.